services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classlink_modal.tsx

```diff
@@ -0,0 +1,111 @@
+export const circleCheckImg = <img alt="" src={circleCheckSrc} />
+
+const TEACHER = 'teacher'
+
+const TEACHER_LACKS_RELEVANT_PREMIUM = 'teacherLacksRelevantPremium'
+const ADMIN_LACKS_RELEVANT_PREMIUM = 'adminLacksRelevantPremium'
+
+const PRIMARY_BUTTON_STYLE = "quill-button contained green medium focus-on-light"
+
+// copy
+const UNLOCK_CLASSLINK_WITH_SCHOOL_OR_DISTRICT_PREMIUM = 'Unlock ClassLink with School or District Premium'
+const SEAMLESSLY_IMPORT_YOUR_CLASSLINK_ROSTERS = 'Seamlessly import your ClassLink rosters.'
+const AUTOMATICALLY_CREATE_AND_SYNC_CLASSLINK_STUDENT_ACCOUNTS = 'Automatically create and sync ClassLink student accounts.'
+const ACCESS_A_HOST_OF_OTHER_PREMIUM_BENEFITS = 'Access a host of other premium benefits, including priority technical support, enhanced reporting, and assistance from our professional learning team.'
+const EXPLORE_PREMIUM = 'Explore premium'
+const CONTACT_YOUR_ADMINISTRATOR = 'Contact your administrator'
+
+
+// links
+const RELATIVE_PREMIUM_HREF = '/premium'
+
+const TEACHER_LACKS_PREMIUM_EMAIL_HREF = `mailto:?subject=Can you help unlock Quill's ClassLink integration?&body=Hi [Insert Your Administrator’s Name],%0D%0A%0D%0A
+I've been digging into the ClassLink integration in Quill, and it seems like a fantastic tool for our teaching needs.
+To tap into its full potential, we'd need a School Premium or District Premium subscription.%0D%0A%0D%0A
+For more details on Quill's premium offerings, they have a page listing the benefits and pricing here:
+https://www.quill.org/premium.%0D%0A%0D%0A
+If you're curious about how Quill’s ClassLink integration works, you can learn more here:
+https://support.quill.org/en/articles/11321582-how-do-i-set-up-a-classlink-rostering-integration-for-my-school-district-for-classlink-quill-administrators.%0D%0A%0D%0A
+I'd appreciate it if you would consider reaching out to Quill at sales@quill.org to discuss the possibility of upgrading our
+subscription. This could open up some exciting opportunities for us.%0D%0A%0D%0A
+Thank you for your time and consideration.%0D%0A%0D%0ABest regards,%0D%0A%0D%0A[Insert Your Name]`
+
+
+const classlinkStates = {
+  [TEACHER_LACKS_RELEVANT_PREMIUM]: {
+    header: UNLOCK_CLASSLINK_WITH_SCHOOL_OR_DISTRICT_PREMIUM,
+    paragraphCopy: 'Looking to streamline your teaching process with Quill’s ClassLink integration? Subscribe to School or District Premium to:',
+    checklistItems: [
+      SEAMLESSLY_IMPORT_YOUR_CLASSLINK_ROSTERS,,
+      AUTOMATICALLY_CREATE_AND_SYNC_CLASSLINK_STUDENT_ACCOUNTS,
+      ACCESS_A_HOST_OF_OTHER_PREMIUM_BENEFITS
+    ],
+    buttons: [
+      <a className={PRIMARY_BUTTON_STYLE} href={RELATIVE_PREMIUM_HREF} key={EXPLORE_PREMIUM} rel="noopener noreferrer" target="_blank">{EXPLORE_PREMIUM}</a>,
+      <a className={PRIMARY_BUTTON_STYLE} href={TEACHER_LACKS_PREMIUM_EMAIL_HREF} key={CONTACT_YOUR_ADMINISTRATOR} rel="noopener noreferrer" target="_blank">{CONTACT_YOUR_ADMINISTRATOR}</a>,
```

Comment:
Am I reading correctly that the only difference between these two is the existence of this second button? This is really just a preference, so take or leave, but if we don't anticipate further differentiating these states in the near term, I would just code the rest of this directly rather than relying on these hashes for all the modal content because it makes it a little trickier to parse -- to see what content is actually displayed I have to look at the render function, then these hashes, then the constants.

Reply:
Good point; there's another modal PR coming. I will re-assess in that one, and simplify if the complexity of this structure is no longer merited.